From 26b12b6afd7b9ab24402774ac9a6c91366a1f005 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Tue, 12 Mar 2019 17:57:26 +0200
Subject: [PATCH 100/235] fsl_qbman: add dummy functions for probe deferal APIs
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add missing definitions for probe deferal APIs in order
to fix these compilation issues for drivers that use them
with the SDK version of the qbman drivers:

drivers/crypto/caam/ctrl.c: In function ‘caam_probe’:
drivers/crypto/caam/ctrl.c:544:9: error: implicit declaration of function ‘qman_is_probed’
drivers/crypto/caam/ctrl.c:554:9: error: implicit declaration of function ‘qman_portals_probed’

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
[Xulin: Original patch taken from NXP LSDK-19.06.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 include/linux/fsl_qman.h | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/include/linux/fsl_qman.h b/include/linux/fsl_qman.h
index 8b169b48d94b..880f320313df 100644
--- a/include/linux/fsl_qman.h
+++ b/include/linux/fsl_qman.h
@@ -3910,6 +3910,16 @@ int qman_p_enqueue_orp(struct qman_portal *p, struct qman_fq *fq,
 int qman_p_enqueue_precommit(struct qman_portal *p, struct qman_fq *fq,
 				const struct qm_fd *fd, u32 flags,
 				qman_cb_precommit cb, void *cb_arg);
+
+static inline int qman_is_probed(void) {
+	return 1;
+}
+
+
+static inline int qman_portals_probed(void) {
+	return 1;
+}
+
 #ifdef __cplusplus
 }
 #endif
-- 
2.17.1

